src/script/interpreter.cpp

```diff
@@ -1484,7 +1485,8 @@ static bool VerifyWitnessProgram(const CScriptWitness& witness, int witversion,
             if (witness.stack.size() != 2) {
                 return set_error(serror, SCRIPT_ERR_WITNESS_PROGRAM_MISMATCH); // 2 items in witness
             }
-            scriptPubKey << OP_DUP << OP_HASH160 << program << OP_EQUALVERIFY << OP_CHECKSIG;
+            temp_script << OP_DUP << OP_HASH160 << program << OP_EQUALVERIFY << OP_CHECKSIG;
+            script = Span<const unsigned char>(temp_script.data(), temp_script.size());
             stack = witness.stack;
```

Comment:
Given that `temp_script` is declared as `const` (see previous reviews comment above), one can simply use `MakeSpan(temp_script)` here.

Reply:
That's unfortunately incorrect. The `script` variable is a Span which doesn't have its own storage, so it needs the `temp_script` to remain intact over its lifetime. I've added a comment to clarify.

Reply:
Whoops, you are of course right, `temp_script` wouldn't be alive anymore when `script` is used below. The evil thing with errors like those is that they are hard to get catch by tests: when I tried out my suggestion all of the unit tests still passed successfully. Obviously even though `temp_script` wasn't alive anymore the data at the address was still there unmodified.
Only creating a destructor `~CScript() { assign(size(), 0}; }` that nullifies the data at the end helped to trigger the failures.